rad/monadic/items.rad

```diff
@@ -118,6 +146,24 @@
         "Comments\n"
         "--------\n\n"
         (pretty-comments cs)))
+      '{:number n
```

Comment:
It's very weird to me to have a single function that is supposed to know how to render every different item comment type. Why not different functions?

Reply:
I agree that items.rad needs some refactoring. In this case it is convenient to have the function `pretty-item-view` handle distinguishing between the item types, otherwise there would be more matching where the function is used. 
Did you have an idea of how to split it into different functions? I can only think of an option to pass flags to the functions of items and depending on that use specific functions, but I am not sure that will improve things.

Reply:
I agree that flags are also not good (even in `pretty-table-header` it looks bad).
I don't entirely understand the "more matching" problem. There'd be a function, `pretty-issue`, that would call `pretty-issue-header`, and so on. You don't need to match on the comment (unless you want to) if you already know at call-site what shape it'll have (which you will since you'll be calling from inside a function that knows the comment shape).

Reply:
But then you would still have to decide what function to call depending on the flag. So for example in `list-items` for each of the following function you would have to match/decide which specific one to call: `pretty-headline`, `pretty-row`, `pretty-table-header`. So would you have them refer to the specific functions depending on the flag?

Reply:
> So for example in list-items for each of the following function you would have to match/decide which specific one to call: pretty-headline, pretty-row, pretty-table-header

I don't understand this. Can you explain?

Here's the rule. If I define a new machine, with it's own set of items, and I don't have write access to this repo, can I re-use any of the functionality? Clearly, right now the answer is "no". That means the abstractions are wrong. Maybe there is no common abstraction, but probably the answer is more of a middle ground: we can't have "do everything" functions such as `pretty-item-view`, but we can have other functions that respect actual invariants. E.g.: 

```
(def prompt-comment (fn [post] ...))
```
Which takes some string `post` to put below the `---`, puts that and the line in a file, and sets you up in an editor, and gives you back the string that was inputed.

Reply:
I agree with you. I think I am just not fully clear on which functions to abstract and which not, yet (fe `list-items`) I think the general direction is clear, but I think it exceeds this PR as it would also change issue and patch code. I can write create an issue for the refactoring.

Reply:
ok

Reply:
Refactoring issue: #665